src/Angor/Shared/WalletOperations.cs

```diff
@@ -128,10 +128,92 @@ public string GenerateWalletWords()
             .SendEstimatedFees(new FeeRate(Money.Coins(sendInfo.FeeRate)));
 
         var signedTransaction = builder.BuildTransaction(true);
-        
+
         return await PublishTransactionAsync(network, signedTransaction);
     }
 
+    private void UpdatePendingLists(AccountInfo accountInfo)
+    {
+        // remove from the pending remove list if it was removed from the indexer
+        var pendingRemove = accountInfo.PendingRemove.ToList();
+        foreach (var utxoData in pendingRemove)
+        {
+            foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+            {
+                if (addressInfo.Address == utxoData.address)
+                {
+                    if (addressInfo.UtxoData.All(_ => _.outpoint.ToString() != utxoData.outpoint.ToString()))
+                    {
+                        accountInfo.PendingRemove.Remove(utxoData);
+                    }
+                }
+            }
+        }
+
+        // remove from the pending add if it was removed from the indexer
+        var pendingAdd = accountInfo.PendingAdd.ToList();
+        foreach (var utxoData in pendingAdd)
+        {
+            foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+            {
+                if (addressInfo.Address == utxoData.address)
+                {
+                    if (addressInfo.UtxoData.Any(_ => _.outpoint.ToString() == utxoData.outpoint.ToString()))
+                    {
+                        accountInfo.PendingAdd.Remove(utxoData);
+                    }
+                }
+            }
+        }
+    }
+
+    public void UpdateAccountInfoWithSpentTransaction(AccountInfo accountInfo, Transaction transaction, string changeAddress)
+    {
+        Network network = _networkConfiguration.GetNetwork();
+
+        var outputs = transaction.Outputs.AsIndexedOutputs();
+        var inputs = transaction.Inputs.Select(_ => _.PrevOut).ToList();
+        
+        foreach (var addressInfo in accountInfo.AddressesInfo.Concat(accountInfo.ChangeAddressesInfo))
+        {
+            // find all spent inputs to mark them as spent
+            foreach (var utxoData in addressInfo.UtxoData)
+            {
+                foreach (var outPoint in inputs)
+                {
+                    if (utxoData.outpoint.ToString() == outPoint.ToString())
+                    {
+                        if (accountInfo.PendingRemove.All(_ => _.outpoint != utxoData.outpoint))
```

Comment:
Don't you need to call to string on the outpoints?

Reply:
good catch